docs/milestone1.md

```diff
@@ -19,7 +19,102 @@ There are several approaches for solving derivatives, but there is a tradeoff be
 
 
 ## How to Use *AwesomeDiff*
+### Installation
+* The package will be available on 'PyPI.' 
+    - You can either install install the package in your local environment or in a virtual environment.  
+* IF you have a Python3 environment with numpy installed ready to go, the awesomediff package can be installed using the following code:
+```
+pip install awesomediff
+```
+* If you want to install the package in a virtual environment, set up the virtual environment using:
+```
+conda create -n awesomediff python=3.6 anaconda
+source activate awesomediff
+git clone https://github.com/awesomediff/cs207-FinalProject.git
+cd cs207-FinalProject
+```
+Then install the dependencies using
+```
+pip install -r requirements.txt
+```
+
+### Usage
+- To use awesomediff, the user should import the library first using code similar to this:
+```python
+import awesomediff as ad
+```
+- Generally, the user should initialize an AutoDiff object first using the module. The AutoDiff object can perform elementary math operations and functions. 
+- The constructor of an AutoDiff object takes in a function as input.
+- The user can calculate the derivative and the output value at a targeting evaluation point.
+
+Below are some example scenarios to demonstrate how the module works:  
+  
+I. Scalar function
+This is the simplest case scenario. The AutoDiff object takes in a function that only involved elementary operations with scalers. In the below example, the AutoDiff object calculates the derivative at 3 for f(x) = 3x+15
+```python
+import awesomediff as ad
+
+# instantiate an AutoDiff object at the given scalar point
+x = ad.variable(5.0)
+
+# create the function needed for differentiation
+scalarFunc = 5*x**2
+
+# get the derivative and output value 
+print(scalarFunc.val)
+print(scalarFunc.der)
```

Comment:
Hwere we access the results with `.val` and `.der` (properties) but below we use `.val()` and `.der()` (methods).  Is this an intentional difference between the scalar case and the vector case?

Or should we just use functions everywhere (which would allow us to give the user options, e.g. `funcVec.der(dimension=2)`)?